### DISCUSSION TITLE
Error handling in Go functions

### DISCUSSION THREAD

**Discussion by mg98** [serverless/serverless#9408] [Q&A]
I have a function with `runtime: go1.x` and an HTTP event (API Gateway trigger).

This is how the implementation for the handler function looks in this case:
```
func Handler(req events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error)
```

This function is called in the main func:
```
func main() {
    lambda.Start(Handler)
}
```

So when the execution arrives at an error, I am returning something like this:
```
return events.APIGatewayProxyResponse{StatusCode: http.StatusBadRequest}, errors.New("missing signature")
```

**What do I expect from this?**

Requests fail with response code 400 and the message "missing signature".

**What do I actually get?**

Requests fail with response code 502 and the `{
    "message": "Internal server error"
}`.

CloudWatch logs the following:
```
missing signature: errorString null
```

...which is also a strange output. It seems to output the type of the error (errorString) and whatever null stands for. Maybe it also makes sense to do it like that, I don't know. But what I want to say with this, in particular with the HTTP response, is that errors are not handled correctly for functions in Go.

**Comment by pgrzesik** [comment]
Hello @mg98 - as this is not related to the Framework, I'm moving it to "Discussions" - this tracker is intended for enhancement proposals and/or bug tracking.
